tasks/utils/emission-disperse-bal.ts

```diff
@@ -0,0 +1,115 @@
+import https from "https"
+import { BN, simpleToExactAmount } from "@utils/math"
+
+const MTA_ADDRESS = "0xF501dd45a1198C2E1b5aEF5314A68B9006D842E0"
+const REPORT_URL = "https://raw.githubusercontent.com/balancer-labs/bal-mining-scripts/master/reports"
+
+interface DisperseDetails {
+    total: BN
+    recipients: Array<string>
+    values: Array<BN>
+}
+interface BalancerReward {
+    address: string
+    amount: string
+}
+interface BalancerDetails {
+    total: BN
+    rewards: Array<BalancerReward>
+}
+
+interface BalancerPolygonReport {
+    disperser: DisperseDetails
+    balancer: BalancerDetails
+}
+
+const log = (message: string, debugMode = false): void => {
+    if (debugMode) console.log(message)
+}
+/**
+ * Download a report report from balancer labs repository, it contains the distribution of rewards that need to be dispersed.
+ *  It downloads the report from the following URL
+ * https://raw.githubusercontent.com/balancer-labs/bal-mining-scripts/master/reports/WEEK/__polygon_0xF501dd45a1198C2E1b5aEF5314A68B9006D842E0.json
+ *
+ * @param {number} report  - Report number from the bal-mining-script repo. eg 79
+ * @param {Object} [options={ debugMode: false }] : If debug mode is set, it logs details about the downloading process.
+ * @return {Promise}  -  {Promise<Array<{ address: string, amount: string }>>}
+ */
+export const fetchBalancerReport = async (
+    report: number,
+    options = { debugMode: false },
+): Promise<Array<{ address: string; amount: string }>> => {
+    const url = `${REPORT_URL}/${report}/__polygon_${MTA_ADDRESS}.json`
+    log(`fetchBalancerReport: get balancer-labs for report ${report}`, options.debugMode)
+    log(`fetchBalancerReport: download report from url :${url}`, options.debugMode)
+
+    return new Promise((resolve, reject) => {
+        https.get(url, (res) => {
```

Comment:
we do have `axios` in the dependencies. That might be easier to use. eg https://github.com/mstable/mStable-contracts/blob/5d57eaecacc0deeb37ecc7b7b6137314da7aed81/tasks/stakingV1.ts#L21

Reply:
resolved with e270e136bd3e196c30033659347192d30ff84da0